lib/shackles/helper_methods.rb

```diff
@@ -0,0 +1,50 @@
+module Shackles
+  module HelperMethods
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+
+    module ClassMethods
+
+      # ======== USAGE =========
+
+      # class SomeModel < ActiveRecord::Base
+      #   def expensive_read_only
+      #     ...
+      #   end
+      #
+      #   shackle_method :expensive_read_only, environment: "slave"
+      #
+      #   def self.class_level_expensive_read_only
+      #     ...
+      #   end
+      #
+      #   shackle_class_method :class_level_expensive_read_only, environment: "slave"
+      # end
+
+
+      def shackle_class_methods(*methods, opts)
+        methods.each { |m| shackle_class_method(m, opts) }
+      end
+
+      def shackle_class_method(method, opts)
+        opts[:target] = self.singleton_class
```

Comment:
passing target as an opt seems odd. why not just call shackle_method on the singleton_class (after extending it if necessary)